addons/mail/models/mail_alias_mixin_optional.py

```diff
@@ -70,6 +70,12 @@ def create(self, vals_list):
                 alias_vals.update(self.env[self._name].with_context(
                     default_alias_domain_id=company.alias_domain_id.id,
                 )._alias_get_creation_values())
+                existing_alias = self.env['mail.alias'].search([
```

Comment:
Moreover it is globally useless, we have a ``_check_unique`` that does more complicated (and complete) check. Do not try to hack low level create please. Globally it is the purpose of the caller to give correct values.

Reply:
I should have asked more before starting this PR, it seems I didn't understand enough about this.

This wasn't meant as a hack really, since I wanted to fix this case of automatic duplicate creation globally, I thought it would be better to do it here, my reasoning was that since the error can happen any time the alias's name is taken from the record's name, the error can occur anywhere in the future.